srcts/src/components/tooltip.ts

```diff
@@ -28,64 +29,73 @@ type UpdateMessage = {
 
   @property({ type: String }) placement: TooltipOptions["placement"] = "auto";
-  @property({ type: String }) options = "{}";
+  @property({ type: String }) bsOptions = "{}";
 
-  get allOptions(): TooltipOptions {
-    const opts = JSON.parse(this.options);
+  private get options(): TooltipOptions {
+    const opts = JSON.parse(this.bsOptions);
     return {
-      title: this.title,
+      title: this.content,
       placement: this.placement,
       // Bootstrap defaults to false, but we have our own HTML escaping
       html: true,
-      sanitize: true,
+      sanitize: false,
       ...opts,
     };
   }
 
-  get title(): string {
-    return (this.children[0] as Element).innerHTML;
+  private get content(): HTMLElement | undefined {
+    return this.children[0] as HTMLElement;
   }
 
+  // The element that triggers the tooltip to be shown
+  private get triggerElement(): ReturnType<typeof getOrCreateTriggerEl> {
+    return getOrCreateTriggerEl(this);
+  }
+
+  // Is the trigger element visible?
+  get visibleTrigger(): boolean {
+    const el = this.triggerElement;
+    return el && el.offsetParent !== null;
+  }
+
+  ///////////////////////////////////////////////////////////////
+  // Methods
+  ///////////////////////////////////////////////////////////////
+
   constructor() {
     super();
     this._onShown = this._onShown.bind(this);
+    this._onInsert = this._onInsert.bind(this);
     this._onHidden = this._onHidden.bind(this);
     this.style.display = "contents";
   }
 
   connectedCallback(): void {
     super.connectedCallback();
-    this.triggerElement.setAttribute("data-bs-toggle", "tooltip");
-    this._tooltip = new Tooltip(this.triggerElement, this.allOptions);
 
-    // This observer watches for changes in the trigger element's visibility
-    // (only when the tooltip is visible). If the trigger element is no longer
-    // visible, then we hide the tooltip (Bootstrap doesn't do this automatically
-    // when programmatically showing a tooltip)
-    this._observer = this._createVisibilityObserver();
+    const trigger = this.triggerElement;
+    trigger.setAttribute("data-bs-toggle", "tooltip");
+    trigger.setAttribute("tabindex", "0");
+    this.tooltip = new bsTooltip(trigger, this.options);
 
-    this.triggerElement.addEventListener("shown.bs.tooltip", this._onShown);
-    this.triggerElement.addEventListener("hidden.bs.tooltip", this._onHidden);
+    this.visibilityObserver = this._createVisibilityObserver();
+    trigger.addEventListener("shown.bs.tooltip", this._onShown);
+    trigger.addEventListener("hidden.bs.tooltip", this._onHidden);
+    trigger.addEventListener("inserted.bs.tooltip", this._onInsert);
   }
 
   disconnectedCallback(): void {
-    this.triggerElement.removeEventListener("shown.bs.tooltip", this._onShown);
-    this.triggerElement.removeEventListener(
-      "hidden.bs.tooltip",
-      this._onHidden
-    );
+    const trigger = this.triggerElement;
```

Comment:
It turns out that we should call `tooltip.dispose()` here to avoid the problem described in #708 

```suggestion
    this.tooltip.dispose();
    const trigger = this.triggerElement;
```